proj-sys/build.rs

```diff
@@ -113,9 +115,32 @@ fn build_from_source() -> Result<std::path::PathBuf, Box<dyn std::error::Error>>
         "cargo:rustc-link-search={}",
         &out_path.join("build/lib").display()
     );
+
     // The PROJ library needs SQLite and the C++ standard library.
     println!("cargo:rustc-link-lib=dylib=sqlite3");
+
+    // On platforms like apples aarch64, users are likely to have installed libtiff with homebrew,
+    // which isn't in the default search path, so try to determine path from pkg-config
+    match pkg_config::Config::new()
+        .atleast_version("4.0")
+        .probe("libtiff-4")
+    {
+        Ok(pk) => {
+            eprintln!(
+                "found acceptable libtiff installed at: {:?}",
+                pk.link_paths[0]
+            );
+            println!("cargo:rustc-link-search=native={:?}", pk.link_paths[0]);
+        }
+        Err(err) => {
+            // pkg-config might not even be installed. Let's try to stumble forward
+            // to see if the build succeeds regardless, e.g. if libtiff is installed
+            // in some default search path.
+            eprintln!("Failed to find libtiff with pkg-config: {}", err);
+        }
+    }
     println!("cargo:rustc-link-lib=dylib=tiff");
```

Comment:
I'm having second thoughts about some of my changes in 9aca22cd849060541e451b035501167dfda70cde 😬 

Assuming that the system proj has tiff was a nice way to avoid building from src in more circumstances, but we're now also requiring libtiff whenever building from source. 

With a little work we might be able to accommodate all these cases...

Reply:
After typing it out, I was motivated to just address this. With https://github.com/georust/proj/pull/95/commits/b0f447446d07cadc2da86d6be3d37eb35c3620d6 we only require tiff when using the `network` feature, which I've now forwarded from proj to proj/sys.

It's *almost* like the old bunled_proj_tiff, except that we currently just assume any pre-existing sytem proj supports tiff.

We *could* add some kind of check t build.rs to make sure the linked libproj supports tiff, but I'm inclined to wait and see if it's actually a problem.

If we wanted to require pkg-config, one (not very impressive) approach would be parsing the output of `pkg-config --libs --static proj` for libtiff.

Reply:
IIRC, kornel's Rust -sys library guide notes that pkg-config can be quite fussy. Homebrew's proj installs pkg-config but of course we can't make that assumption w/r/t/ Linux, so if we can do without it that's great.